mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -474,7 +474,7 @@ OpFoldResult AddOp::fold(ArrayRef<Attribute> operands) {
   auto resultTy = getType().dyn_cast<RankedTensorType>();
   if (!lhsTy || !rhsTy || !resultTy)
     return {};
-  if (lhsTy != rhsTy)
+  if (lhsTy.getRank() != rhsTy.getRank())
```

Comment:
I think you don't need to check that, it would be an invalid tosa op if the [ranks don't match](https://www.mlplatform.org/tosa/tosa_spec.html#_add) (so the verifier should catch this).